public/app/plugins/datasource/cloud-monitoring/components/VisualMetricQueryEditor.tsx

```diff
@@ -38,12 +40,24 @@ function Editor({
   const [labels, setLabels] = useState<{ [k: string]: any }>({});
   const { projectName, groupBys, crossSeriesReducer } = query;
   const metricType = getMetricType(query);
+  const [timeState, setTime] = useState<{ from: string; to: string } | null>(null);
+  const timeSrv = getTimeSrv();
+
+  const useTime = (time: { from: string; to: string }) => {
+    if (timeState !== null && (timeState.from !== time.from || timeState.to !== time.to)) {
+      setTime({ ...timeSrv.time });
+    } else if (!timeState) {
+      setTime({ ...timeSrv.time });
+    }
```

Comment:
what if rather than this `else` you do in the initialization:

```
  const timeSrv = getTimeSrv();
  const [timeState, setTime] = useState<TimeOption>(timeSrv.time);
```
?